code/modules/projectiles/projectile/beams.dm

```diff
@@ -869,3 +866,127 @@ var/list/beam_master = list()
 		return 1
 	else
 		return ..()
+
+/obj/item/projectile/beam/bison/heal
+	name = "heal ray"
+	damage = 0
+	icon = 'icons/obj/lightning.dmi'
+	icon_state = "heal"
+	fire_sound = null
+	custom_impact = 1
+	var/found_target = 0
+
+/obj/item/projectile/beam/bison/heal/hurt
+	icon_state = "hurt"
+
+/obj/item/projectile/beam/bison/heal/process()
```

Comment:
I don't think the other copypastes are as bad but this one is pretty unforgiveable